### ISSUE TITLE
Merge Agent with API

### ISSUE THREAD

**Issue by itpetey** [intecture/api#15]
The Agent provides endpoints for the API that are routable via a ZeroMQ pipe. However all the metadata required to autogenerate endpoints and serialize the data for transit is contained within the API.

In order to reduce duplication and unnecessary admin overhead, a parser should be built to interpret API endpoints and autogenerate the required stubs (`agent_endpoints.rs`). A Rust module, `agent.rs`, would be the binary target responsible for initialising ZMQ sockets and starting the Agent service.

This issue is blocked by https://github.com/Intecture/api/issues/14.

**Comment by itpetey** [comment]
Change:

```
let mut host = Host::new();
host.connect(...);
```

to:

```
let mut host = Host::connect(...);
// or
let mut host = Host::local();
```

**Comment by itpetey** [comment]
Fixed by 57d3b7e.
